src/Cli/dotnet/CommonOptions.cs

```diff
@@ -36,7 +36,7 @@ internal static class CommonOptions
                 description)
             {
                 Argument = new Argument<string>(CommonLocalizableStrings.FrameworkArgumentName)
-                    .AddSuggestions(Suggest.TargetFrameworksFromProjectFile().ToArray())
+                    .AddSuggestions(Suggest.TargetFrameworksFromProjectFile())
```

Comment:
Now that methods such as `Suggest.TargetFrameworksFromProjectFile` return delegates, when do those delegates get executed?  From your before/after numbers, there are certainly big improvements, so my suspicion is that for at least some of these cases, we don't execute this code.

Reply:
Yes, we don't always execute the code now, we only call the delegates when the user 'asks' for option suggestions, so when `dotnet [suggest]` or `dotnet complete` are called, or during tab completion. One new thing with System.CommandLine is that help also displays option suggestions, so some of the perf impact has shifted to commands like `dotnet build -h` (see my note in the description).

Reply:
Awesome.  One follow-up: I notice that the time in `dotnet build -h` came down from 2 seconds to 0.8 seconds.  Is there a functionality change there, or is it just that it doesn't have to do as much work to compute suggestions for this command, where before, we computed all suggestions synchronously regardless of command?

Reply:
No functional change, we're just only computing suggestions for build options, whereas before we were computing them for all commands, not just build.

Reply:
Awesome.  Thanks!